presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RewriteAggregationIfToFilter.java

```diff
@@ -231,17 +250,22 @@ private boolean shouldRewriteAggregation(Aggregation aggregation, ProjectNode so
         return expression.getForm() == IF && Expressions.isNull(expression.getArguments().get(2));
     }
 
-    private boolean canUnwrapIf(SpecialFormExpression ifExpression)
+    private boolean canUnwrapIf(SpecialFormExpression ifExpression, AggregationIfToFilterRewriteStrategy rewriteStrategy)
     {
         // Some use cases use IF expression to avoid returning errors when evaluating the true branch. For example, IF(CARDINALITY(array) > 0, array[1])).
         // We shouldn't unwrap the IF for those cases.
         // But if the condition expression doesn't reference any variables referenced in the true branch, unwrapping the if should not cause exceptions for the true branch.
         Set<VariableReferenceExpression> ifConditionReferences = VariablesExtractor.extractUnique(ifExpression.getArguments().get(0));
         Set<VariableReferenceExpression> ifResultReferences = VariablesExtractor.extractUnique(ifExpression.getArguments().get(1));
         if (ifConditionReferences.stream().noneMatch(ifResultReferences::contains)) {
+            // The IF expression can be safely unwrapped, ignore the rewrite strategy.
             return true;
         }
 
+        if (rewriteStrategy == AggregationIfToFilterRewriteStrategy.FILTER_WITH_IF) {
+            return false;
+        }
+
```

Comment:
Could you add test where the two modes behave differently? such as:
IF(a > 1, a)